File: docs/js/Core0/Service-Desk-1/simulation.js
/*******************************************************
 Simulation Scenario Settings
********************************************************/
sim.scenario.durationInSimTime = 1000;
/*******************************************************
 Simulation Model
********************************************************/
sim.model.time = "discrete";
sim.model.timeUnit = "min";  // minutes
sim.model.objectTypes = ["ServiceDesk"];
sim.model.eventTypes = ["CustomerArrival", "CustomerDeparture"];
/*******************************************************
 Initial State
********************************************************/
sim.scenario.setupInitialState = function () {
  // Create initial objects
  var sD = new ServiceDesk({id: 1, queueLength: 0});
  // Schedule initial events
  sim.FEL.add( new CustomerArrival({occTime:1, serviceDesk: sD}));
}
/*******************************************************
 Statistics variables
********************************************************/
sim.model.setupStatistics = function () {
  sim.stat.arrivedCustomers = 0;
  sim.stat.departedCustomers = 0;
  sim.stat.maxQueueLength = 0;
};
/*******************************************************
 Define an experiment (type)
********************************************************/
/*
sim.experimentType = {
  title: `Simple Experiment with 10 replications, each running for ${sim.scenario.durationInSimTime} ${sim.model.timeUnit}.`,
  nmrOfReplications: 10
};
*/

File: Python/Core1/workstation_1/simulation.py
import sys, os

module_path = os.path.abspath('../framework/')
sys.path.insert(1, module_path)
from simulator import Simulator, Model

from workstation import WorkStation
from part_arrival import PartArrival

# create simulator (object)
sim = Simulator()
#*******************************************************
#*** Simulation Model  *********************************
#*******************************************************
timeModel = "continuous"
timeUnit = "min"
objectTypes = ["WorkStation"]
eventTypes = ["PartArrival", "ProcessingStart", "ProcessingEnd"]
sim.model = Model( timeModel, timeUnit, objectTypes, eventTypes)
#*******************************************************
#*** Simulation Scenario/Configuration Settings ********
#*******************************************************
sim.scenario.durationInSimTime = 20*60  # 20 hours
sim.config.createLog = True
#*******************************************************
#*** Initial State *************************************
#*******************************************************
def setupInitialState():
    ws = WorkStation( sim, 1, "ws1")
    sim.FEL.add( PartArrival( sim, ws, occTime=1))
#*******************************************************
#*** Statistics Variables ******************************
#*******************************************************
def setupStatistics():
    sim.stat = {"arrivedParts": 0, "departedParts": 0, "maxQueueLength": 0}
#************************************************************
#*** Overwriting the (Abstract) Methods of Scenario and Model
#************************************************************
sim.scenario.setupInitialState = setupInitialState
sim.model.setupStatistics = setupStatistics
#********************************************************
# Running the Standalone Scenario                       *
#********************************************************
# sim.runStandaloneScenario()
experiment_type = {
    "id": 0,
    "title": f"Simple Experiment with 10 replications, each running for {sim.scenario.durationInSimTime} {sim.model.timeUnit}.",
    "nmrOfReplications": 10,
    "seeds": [123, 234, 345, 456, 567, 678, 789, 890, 901, 1012],
    "storeExpResults": True 
}

sim.experiment_type = experiment_type  # Set the experiment_type attribute in the Simulator instance
sim.runSimpleExperiment() 